packages/indexer-agent/src/agent.ts

```diff
@@ -137,11 +182,34 @@ class Agent {
     )
 
     const indexingRules = timer(60_000).tryMap(
-      () => this.indexer.indexingRules(true),
+      async () => {
+        let rules = await this.indexer.indexingRules(true)
+        const subgraphRuleIds = rules
+          .filter(
+            rule => rule.identifierType == SubgraphIdentifierType.SUBGRAPH,
+          )
+          .map(rule => rule.identifier!)
+        const subgraphsMatchingRules = await this.network.subgraphs(
+          subgraphRuleIds,
+        )
+        if (subgraphsMatchingRules.length >= 1) {
+          const epochLength =
+            await this.network.contracts.epochManager.epochLength()
+          const blockPeriod = 15
+          const bufferPeriod = epochLength.toNumber() * blockPeriod * 100 // 100 epochs
```

Comment:
Probably going to want to parameterize this, so indexers can set the period for which they continue to support the previous version.